sdk/core/azure_core_test/src/recording.rs

```diff
@@ -116,13 +116,17 @@ impl Recording {
     ///     let recording = ctx.recording();
     ///
     ///     let mut options = MyClientOptions::default();
-    ///     ctx.instrument(&mut options.client_options);
+    ///     ctx.instrument(&mut options.client_options, None);
     ///
     ///     let client = MyClient::new("https://azure.net", Some(options));
     ///     client.invoke().await
     /// }
     /// ```
-    pub fn instrument(&self, options: &mut ClientOptions) {
+    pub fn instrument(
+        &self,
+        options: &mut ClientOptions,
+        recording_options: Option<RecordingOptions>,
```

Comment:
I'd like to see this more like our client methods and I admit I should've done that initially. So `RecordingOptions` should be `InstrumentOptions` and have an embedded `client_options: ClientOptions` (not `Option`). You could either put the `remove_header` field in there inline, or embed `RecordingOptions` as-is if that makes more sense, but since it's not a leaf node it should not be optional - just the fields.